### ISSUE TITLE
There should be a dummy tag that Doxygen ignores (Origin: bugzilla #164073)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#1413]
status NEW severity *enhancement* in component *general* for *---*
Reported in version *unspecified* on platform *Other*
Assigned to: Dimitri van Heesch

On 2005-01-14 13:09:33 +0000, Georg Schwab wrote:

    I use the UML tool Together from Borland, which uses its own java-doc tags 
    (e.g. @stereotype and @hidden) for UML informaton. If I had a tag (e.g. 
    \ignore) that tells Doxygen to ignore the rest of the line, then I could
    create
    aliases to skip this extra information as follows:
    
    ALIASES = stereotype=\ignore hidden=\ignore

On 2015-02-05 16:37:29 +0000, simon.watts@physics.org wrote:

    Define ALIASES for the Borland tags which expand to whitespace.

**Comment by Krinkle** [comment]
The code base I'm indexing is https://github.com/wikimedia/mediawiki. There are a number of annotations in our source code that are for other workflow tools and have no relation to the documentation.

For example, `@suppress` to acknowledge a false-negative from the static analysis tool Phan for PHP:

```php
class AuthenticationRequest {
	/**
	 * Merge the output of multiple getFieldInfo() calls.
	 *
	 * @param AuthenticationRequest[] $reqs
	 * @return array
	 * @suppress PhanTypeInvalidDimOffset
	 */
	public static function mergeFieldInfo( array $reqs ) {
```

Or, `@method` to virtually defined a more concrete definition of a method for a static analysis tool:

```php
/**
 * @since 1.29
 * @method ChangesListBooleanFilter[] getFilters()
 */
class ChangesListBooleanFilterGroup extends ChangesListFilterGroup {
```

These currently result in output like the following:
<img width="671" alt="Screenshot" src="https://user-images.githubusercontent.com/156867/64899884-7e54ec80-d685-11e9-8530-065c8411c139.png">

I've tried various different things in the hopes it would help exclude or ignore this from the output, but none of these worked out.

```
ALIASES                = "method=" \
```
```
ALIASES                = "method= " \
```
```
ALIASES                = "method=\n" \
```
```
ALIASES                = "method{1}=" \
```

**Comment by albert-github** [comment]
Your attempts will indeed not help as they only remove the command in the first 3 cases, an leave the text following it.
The `method{1}` version does not work either as it needs the text in a form like `{...}` and you don't have the curly brackets.

The solution I see at the moment is to write an input filter where you remove the relevant parts of the text.

Some further thoughts:
- where should the command end (i.e. next word, end of line, end of paragraph)?
- how to handle other commands "inside" such commands (e.g `@suppress suppress @b this in doxygen`)?

**Comment by Krinkle** [comment]
@albert-github The way I think about annotations (for PHP code) is that only the start of the line can begin a command, except if the current command prints the values and supports free-form text with markdown and/or nested commands.

This would mean that something like `@internal` should not do anything if it appears in the middle of another command's value because it is logically a "block" level command, not suitable for nesting.

```
/**
 * @param {string} $arg1 The text here supports nestable annotations and Markdown, but
 * a command line @private or @internal or @param should not mean anything here.
 */
```

Overall I think it's probably okay not to have that logic in Doxygen, because we can always escape it where needed.

But in that case, the alternative is to have a way for the command to have full control of the value buffer. For example, if the logic for `@mydummy` does not print anything (e.g. a dummy, or like `{..}` without printing `\1` anywhere), then if you use `@mydummy foo @private bar` should be completely omitted. In other words, even if it triggers `@private`, the processing if that would somehow be buffered and `@mydummy` would still be in control of whether to output it or not.

In any event, yeah, we need a way to say that a command should be ignored, and probably also a way to say if it is inline, line, or multi-line. But maybe multi-line would work as default, e.g. everything until another line starts with a command.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7264

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by hashar** [comment]
MediaWiki task for later reference https://phabricator.wikimedia.org/T233788

**Comment by hashar** [comment]
I thought about aliasing those to `\internal` but it only closes at the end of the code block, not at the end of line. So given `INTERNAL_DOCS = NO` and:
```
ALIASES+suppress=\internal
```
A code such as:
```
/**
 * @suppress SomeLinterError
 * @return bool
 */
```
Would cause `@return bool` to be flagged internal and thus not included.

I then though of using an alias to wrap the content:
```
ALIASES+=suppress[1]=\internal\1\endinternal
```

But the alias tokens are space delimited. Thus whenever the command is passed more than one argument, the remaining arguments will not be wrapped and still be shown.

We would need a command that wrap until end of line, or have some syntax for ALIASES to support any arguments until end of line. For example `*`:
```
ALIASES+=suppress[*]=\internal\*\internal
```


Then I guess `\noop` is good enough :)

**Comment by albert-github** [comment]
`\noop` does work on a line so should work and you can specify `ALIASES += suppress=@noop`.

Note: `ALIASES` are wit curly brackets (`{ ...}`) not with square brackets (`[ ... ]`).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
